### PR TITLE
DHistogram clean-up: remove unused parameter nbin_cats

### PR DESCRIPTION
**Pull request by michalkurka**
The main goal of this clean-up is to remove nbin_cats from method signatures - it was not actually used and the source of the parameter was parms._nbin_cats instead.

### CODE REVIEW

File: h2o-algos/src/main/java/hex/tree/ScoreBuildHistogram2.java [hunk 1]
```diff
@@ -299,7 +296,7 @@ public ComputeHistoThread makeCopy() {
     }
 
     @Override
-    protected void map(int id){
+    protected void map(int UNUSED){
```
Comment (maurever) [inline h2o-algos/src/main/java/hex/tree/ScoreBuildHistogram2.java:299]: This change looks weird. I rather keep it as `id` if there is no solution for removing this from the parent class.

**Review by maurever — COMMENTED** [review]
I am surprised how much dead code is in this part of the tree calculation.  I checked everything, and it looks really unused. But I would rather wait for tests to be sure.

File: h2o-algos/src/main/java/hex/tree/ScoreBuildHistogram2.java (hunk 1 above)
Comment (michalkurka) [reply h2o-algos/src/main/java/hex/tree/ScoreBuildHistogram2.java:299]: ok, sure - will remove

**Comment by michalkurka** [comment]
@maurever tests look good - can you please approve? I will remove the problematic commit you pointed out

**Review by maurever — APPROVED** [review]
Sure, thank you @michalkurka!

**Comment by michalkurka** [comment]
fixed, thank you @maurever

**Pull request merged**